src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -164,8 +165,12 @@ int replicationFactor() {
     public void run() {
       try {
         if (_topicCreationEnabled) {
-          _topicFactory.createTopicIfNotExist(_zkConnect, _topic, _configuredTopicReplicationFactor,
-              _partitionsToBrokerRatio, new Properties());
+          try {
+            _topicFactory.createTopicIfNotExist(_zkConnect, _topic, _configuredTopicReplicationFactor,
+                _partitionsToBrokerRatio, new Properties());
+          } catch (TopicExistsException e) {
```

Comment:
Topic factories should catch this exception. A method called createTopicIfNotExist() kind of implies that you should be free to call this method without getting exceptions if the topic already exists.